### DISCUSSION TITLE
Couldn't run e2e tests in parallel NestJS

### DISCUSSION THREAD

**Discussion by kevindavee** [mikro-orm/nestjs#97] [Q&A]
**Describe the bug**
I want to run my jest e2e test in parallel. I run the migration on the App module `onModuleInit`.  I initialized the Testing Module in `beforeAll` of my test file. I didn't run the jest test using `--runInBand` because in the CI, I would like to have the test running in parallel. But I got `UniqueConstraintViolationException: create database "test" - duplicate key value violates unique constraint "pg_database_datname_index"`. Am I setting up the e2e test environment correctly?

```typescript
app.module.ts

async onModuleInit(): Promise<void> {
  await this.orm.getMigrator().up();
}
```

```typescript
*.e2e-spec.ts

beforeAll(async () => {
  const module: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
   }).compile();
  const app = module.createNestApplication();
  await app.init()
});
```

**To Reproduce**
1. Have the migrator run in `onModuleInit`
2. Have more than 1 test file.

**Expected behavior**
I wanted to have my e2e tests running in parallel, and have my test database teared down everytime the test is finished.

**Versions**

| Dependency | Version |
| - | - |
| node | 18 |
| typescript | 4.3.5 |
| mikro-orm | 5.6.1 |
| postgres | 5.6.1 |

**Comment by B4nan** [comment]
You either need to implement some locking in the `onModuleInit` method, so you ensure the migrations are never executed in parallel, or just run the migrations first before you run the tests. Or have a dedicated database for each test, that is the most correct solution to me if you want to test things on a real database in parallel.

**Reply by kevindavee** [reply]
A dedicated database for each test is not feasible. If I want to run in parallel using ten workers, I have to run 10 DB simultaneously. I'll try to run the migrations first before I run the tests. I think that's the most feasible so far.
